udev/udev-qubes-block.rules

```diff
@@ -10,7 +10,10 @@ KERNEL=="xvda|xvdb|xvdc*|xvdd", ENV{UDISKS_IGNORE}="1"
 # Skip xen-blkfront devices
 ENV{MAJOR}=="202", GOTO="qubes_block_end"
 
-# skip devices excluded elsewhere
+# Do not expose raw encrypted block devices (but any sub-partitions are fair game)
+ENV{DM_UUID}=="CRYPT-PLAIN-*", ENV{UDISKS_IGNORE}="1", GOTO="qubes_block_end"
```

Comment:
Won't this exclude also encrypted USB sticks (or those already opened via `cryptsetup open`)? If the purpose is skipping swap, maybe use something that identify swap directly (the device it uses for example).

Reply:
We don’t know what the device will be used for yet, but I can specifically specify `CRYPT-PLAIN-swap`.  That said, in dom0 we should probably use the broader rule, to account for someone needing to expose a LUKS device to a qube.